models/models.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <pybind11/pytypes.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/numpy.h>
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/labeled_graph.hpp>
+#include <boost/graph/breadth_first_search.hpp>
+#include <boost/pending/indirect_cmp.hpp>
+#include <boost/range/irange.hpp>
+#include <boost/archive/text_iarchive.hpp>
+#include <boost/archive/text_oarchive.hpp>
+#include <boost/variant.hpp>
+
+#include <string>
+#include <map>
+
+// Typedefs for tag hierarchy
+using NodeType =
+    std::map<std::string, boost::variant<bool,
+    int,
+    double,
+    pybind11::none,
+    std::string,
+    std::vector<int>,
+    std::vector<std::string>>>;
+
+
+struct Modelhierarchy {
+    std::string id;
+    NodeType properties;
+};
+
+struct Connection {
```

Comment:
Should this not have been a typedef?

Reply:
It needs to be a struct because that is how we can associate data with edges in the boost graph library. Can add a comment about this, as it serves the same purpose as Modelhierarchy struct commented above.